pkg/geo/parse.go

```diff
@@ -25,6 +26,122 @@ import (
 	"github.com/twpayne/go-geom/encoding/wkb"
 )
 
+// Threshold to assume that both float64 values are
+// equivilent
+const float64EquivalentThreshold = 1e-9
+
+// longitudeNormalize will normalize angles to
+// spherical polar angles (-180 to 180)
+func longitudeNormalize(log float64) float64 {
+	if log > 360 {
+		log = math.Remainder(log, 360)
+	}
+	if log < -360 {
+		log = math.Remainder(log, -360)
+	}
+	if log > 180 {
+		log += -360
+	}
+	if log < -180 {
+		log += 360
+	}
+	if log == -180 {
+		return 180
+	}
+	if log == -360 {
+		return 0
+	}
+	return log
+}
+
+// latitudeNormalize will normalize angles to
+// spherical azimuth angle (-90 to 90)
+func latitudeNormalize(lat float64) float64 {
+	if lat > 360 {
+		lat = math.Remainder(lat, 360)
+	}
+	if lat < -360 {
+		lat = math.Remainder(lat, -360)
+	}
+	if lat > 180 {
+		lat = 180 - lat
+	}
+	if lat < 180 {
+		lat = -180 - lat
+	}
+	if lat > 90 {
+		lat = 180 - lat
+	}
+	if lat < -90 {
+		lat = -180 - lat
+	}
+
+	return lat
+}
+
+func makeValidPoint(lat float64, log float64) (float64, float64) {
+	if math.IsNaN(lat) || math.IsNaN(log) {
+		return lat, log
+	}
+	_lat := latitudeNormalize(lat)
+	_log := longitudeNormalize(log)
+
+	// Normalize occurs in integer part of lat/log
+	// if the difference are within 1 we could set the original lat/log
+	// to prevent unwanted float point arithmetic errors
+	if math.Abs(lat-_lat) <= 1 && math.Abs(log-_log) <= 1 {
+		return lat, log
+	}
+	return _lat, _log
+}
+
+func makeGeomValid(t geom.T) {
+	if t.Layout() != geom.XY {
+		return
+	}
+
+	switch repr := t.(type) {
+	case *geom.Point:
+		y, x := makeValidPoint(repr.Y(), repr.X())
```

Comment:
you can just iterate over `g.FlatCoords()`, except for `geom.GeometryCollection` (see how boundingBoxForGeomT works).